src/staveconnector.ts

```diff
@@ -30,6 +30,13 @@ function drawBoldDoubleLine(ctx: RenderContext, type: number, topX: number, topY
   ctx.fillRect(topX - thickLineOffset, topY, variableWidth, botY - topY);
 }
 
+/**
+ * see {@link StaveConnector.type} & {@link StaveConnector.typeString}
+ */
+export type StaveConnectorType =
+  | typeof StaveConnector['type'][keyof typeof StaveConnector['type']]
+  | keyof typeof StaveConnector['typeString'];
```

Comment:
I like this PR, but I'm having a hard time grokking this type definition. @ronyeh thoughts on this?

Reply:
@0xfe --just pushed a revision that makes the StaveConnectorType definition cleaner and sets the types of StaveConnector.type and StaveConnector.typeString by using the `StaveConnectorType` definition, rather than the other way around. Hopefully it's a lot more readable/grokkable.

Reply:
Thanks @gristow -- this is definitely better. Can you add a comment explaining this type so future me understands it :-)